sympy/assumptions/handlers/ntheory.py

```diff
@@ -49,8 +49,12 @@ def Mul(expr, assumptions):
                 pass
             else:
                 break
+        for arg in expr.args:
+            if arg.is_number and arg.is_composite:
+                return False
+            else:
+                return None
         else:
-            # a product of integers can't be a prime
             return False
```

Comment:
Is this correct? It seems that this will be executed if there are no numbers, only symbols.